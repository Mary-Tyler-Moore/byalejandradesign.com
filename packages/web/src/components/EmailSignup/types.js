/** @flow */
export type State = {
  email: string,
};
