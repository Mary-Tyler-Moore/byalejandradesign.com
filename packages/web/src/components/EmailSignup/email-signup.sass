@import ../../sass/global.sass

.emailSignup
  width: 100%

  &_label
    margin: 0

    @extend %subtitle-copy-2

  &_input
    width: 100%
    margin-bottom: 1rem
    padding: 0.5rem
    border-radius: 0.25rem
    border: none

    @extend %body-copy-2

    margin: 1.5rem 0

    &::placeholder
      font-style: italic
      font-weight: 700

    &:focus
      outline: none
