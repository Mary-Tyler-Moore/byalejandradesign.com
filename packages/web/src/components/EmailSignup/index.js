/** @flow */
export { default } from './EmailSignup'
export * from './types'

