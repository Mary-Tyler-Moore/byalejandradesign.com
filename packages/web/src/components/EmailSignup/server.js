import axios from 'axios';
import { MAIL_DOMAIN, API_KEY } from '@byalejandradesign/server-env';

const server = axios.create({
  baseURL: MAIL_DOMAIN,
  headers: {
    Authorization: API_KEY,
  },
});

export default server;
