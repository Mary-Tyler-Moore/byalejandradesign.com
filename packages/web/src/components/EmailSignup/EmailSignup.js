/** @flow */
import * as React from 'react';
import { Form } from '@njmyers/component-library';
import Button from '../Button';
// types
import type { State } from './types';
// styles
import './email-signup.sass';

class EmailSignup extends React.Component<{}, State> {
  state = {
    email: '',
  };

  onChange = (e: SyntheticEvent<HTMLInputElement>) => {
    e.preventDefault();

    const {
      currentTarget: { value },
    } = e;

    this.setState((state) => ({
      email: value,
    }));
  };

  onSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    e.preventDefault();

    this.setState((state) => ({ email: '' }));
  };

  render() {
    return (
      <form name="emailSignup" className="emailSignup" onSubmit={this.onSubmit}>
        <Form.Input
          block="emailSignup"
          name="email"
          label="Signup For Our Newsletter"
          type="email"
          value={this.state.email}
          onChange={this.onChange}
          required
        />
        <Button type="submit">Sign Up</Button>
      </form>
    );
  }
}

export default EmailSignup;
