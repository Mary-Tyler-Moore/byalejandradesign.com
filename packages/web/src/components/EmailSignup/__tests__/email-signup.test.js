import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import EmailSignup from '../EmailSignup';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<EmailSignup />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});


