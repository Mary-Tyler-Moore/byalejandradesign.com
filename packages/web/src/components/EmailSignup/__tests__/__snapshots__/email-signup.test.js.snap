// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `
<form
  className="emailSignup"
  name="emailSignup"
  onSubmit={[Function]}
>
  <div
    className="emailSignup"
  >
    <label
      className="emailSignup_label"
      element="label"
      htmlFor="email"
    >
      Signup For Our Newsletter
    </label>
    <input
      className="emailSignup_input"
      element="input"
      name="email"
      onChange={[Function]}
      placeholder=""
      required={true}
      type="email"
      value=""
    />
  </div>
  <button
    className="button-yellow"
    style={
      Object {
        "margin": "0",
        "width": "inherit",
      }
    }
    type="submit"
  >
    <span>
      Sign Up
    </span>
  </button>
</form>
`;
